management-service/src/main/java/pl/pwr/zpi/reports/service/ReportGenerationService.java

```diff
@@ -3,35 +3,35 @@
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
+import pl.pwr.zpi.cluster.repository.ClusterRepository;
 import pl.pwr.zpi.notifications.ReportNotificationService;
 import pl.pwr.zpi.reports.broker.ReportPublisher;
 import pl.pwr.zpi.reports.dto.event.ReportGenerated;
 import pl.pwr.zpi.reports.dto.event.ReportRequestFailed;
 import pl.pwr.zpi.reports.dto.event.ReportRequested;
 import pl.pwr.zpi.reports.dto.request.CreateReportRequest;
+import pl.pwr.zpi.reports.dto.request.CreateScheduleRequest;
+import pl.pwr.zpi.reports.dto.scheduler.ClusterSchedule;
 import pl.pwr.zpi.reports.entity.report.Report;
-import pl.pwr.zpi.reports.entity.report.node.NodeIncident;
 import pl.pwr.zpi.reports.entity.report.request.ReportGenerationRequestMetadata;
 import pl.pwr.zpi.reports.enums.ReportGenerationStatus;
 import pl.pwr.zpi.reports.repository.*;
 
-import java.util.List;
-
 @Slf4j
 @Service
 @RequiredArgsConstructor
 public class ReportGenerationService {
 
     private final ReportPublisher reportPublisher;
     private final ReportNotificationService reportNotificationService;
-
     private final ReportRepository reportRepository;
     private final NodeIncidentRepository nodeIncidentRepository;
     private final NodeIncidentSourcesRepository nodeIncidentSourcesRepository;
     private final ApplicationIncidentRepository applicationIncidentRepository;
     private final ApplicationIncidentSourcesRepository applicationIncidentSourcesRepository;
-
     private final ReportGenerationRequestMetadataRepository reportGenerationRequestMetadataRepository;
+    private final SchedulerRepository schedulerRepository;
```

Comment:
I'd suggest creating a ReportScheduleService as Scheduling is not strictly related to report generation and it should only make use of its API